src/main/java/org/truffleruby/interop/InteropNodes.java

```diff
@@ -833,6 +835,26 @@ protected ForeignToRubyNode createForeignToRubyNode() {
 
     }
 
+    @CoreMethod(names = "to_java_array_inner", isModuleFunction = true, required = 1)
+    public abstract static class InteropToJavaArrayNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization(guards = "isRubyArray(array)")
+        public Object toJavaArray(DynamicObject array) {
+            return JavaInterop.asTruffleObject(ArrayStrategy.of(array).newMirror(array).copyArrayAndMirror().getArray());
+        }
+
+    }
+
+    @CoreMethod(names = "deproxy", isModuleFunction = true, required = 1)
+    public abstract static class DeproxyNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
```

Comment:
A guard with `JavaInterop.isJavaObject` would provide a better error in case it's not a JavaObject.